parser-typechecker/src/Unison/Codebase/Init.hs

```diff
@@ -124,21 +118,19 @@ createCodebase cbInit debugName path = do
 
 -- previously: initCodebaseOrExit :: CodebasePath -> m (m (), Codebase m v a)
 -- previously: FileCodebase.initCodebase :: CodebasePath -> m (m (), Codebase m v a)
-openNewUcmCodebaseOrExit :: MonadIO m => Init m Symbol Ann -> DebugName -> CodebasePath -> m (m (), Codebase m Symbol Ann)
-openNewUcmCodebaseOrExit cbInit debugName path = do
+withNewUcmCodebaseOrExit :: MonadIO m => Init m Symbol Ann -> DebugName -> CodebasePath -> (Codebase m Symbol Ann -> m r) -> m r
+withNewUcmCodebaseOrExit cbInit debugName path action = do
   prettyDir <- P.string <$> canonicalizePath path
-  createCodebase cbInit debugName path >>= \case
+  let codebaseSetup codebase = do
+        liftIO $ PT.putPrettyLn' . P.wrap $ "Initializing a new codebase in: " <> prettyDir
+        Codebase.installUcmDependencies codebase
+  createCodebase cbInit debugName path (\cb -> codebaseSetup cb *> action cb) >>=
```

Comment:
Had to rearrange things just a little here so we could properly call the codebase setup in the action handler.